documentation/general/design/dash-sonic-hld.md

```diff
@@ -0,0 +1,467 @@
+
+In the outbound flow, the appliance shall assume it is the first appliance to apply policy. It applies the outbound ACLs in three stages (VNIC, Subnet and VNET), processed in order, with the outcome being the most restrictive of the three ACLs combined. 
+
+After the ACL stage, it does LPM routing based on the inner dst-ip and applies the respective action (encap, subsequent CA-PA mapping). Finally, update the connection tracking table for both inbound and outbound. 
+	
+## 2.2 Inbound packet processing pipeline
+	
+   ![dash-intbound](https://github.com/prsunny/DASH/blob/main/Assets/pkt_flw_inbound.png)
+
+Based on the incoming packet's VNI, if it does not match against any reserved VNI, the pipeline shall set the direction as RX(Inbound) and using the inner dst-mac, maps to the corresponding ENI. In the inbound flow, Routing (LPM) lookup happens first based on the inner dst-ip and does a CA-PA validation based on the mapping. After LPM is the three stage ACL, processed in order. ACLs can have multiple src/dst IP ranges or port ranges as match criteria.
+	
+It is worth noting that CA-PA mapping table shall be used for both encap and decap process
+	
+# 3 Modules Design
+
+The following are the schema changes. The NorthBound APIs shall be defined as sonic-yang in compliance to [yang-guideline](https://github.com/Azure/SONiC/blob/master/doc/mgmt/SONiC_YANG_Model_Guidelines.md)
+
+For DASH objects, the proposal is to have its own DB instance (DASH_APP_DB). Ref [link](https://github.com/Azure/sonic-buildimage/blob/master/dockers/docker-database/database_config.json.j2), thus ensuring isolation from regular configs that are persistent in the appliance across reboots. All the DASH objects are programmed by SDN and hence treated differently from the existing Sonic L2/L3 'switch' DB ojects 
+
+```
+        "DASH_APP_DB" : {
+            "id" : 15,
+            "separator": ":",
+            "instance" : "redis"
+        }
+	
+	"DASH_STATE_DB" : {
+            "id" : 16,
+            "separator": "|",
+            "instance" : "redis"
+        }
+```
+
+## 3.1 Config DB
+
+### 3.1.1 DEVICE Metadata Table
+
+```
+"DEVICE_METADATA": {
+    "localhost": {
+        "subtype": "appliance",
+        "type": "sonichost",
+        "switch_type": "dpu",
+        "sub_role": "None"
+     }
+}
+```
+
+### 3.1.2 VXLAN Table
+
+```
+VXLAN_TUNNEL|{{tunnel_name}} 
+    "src_ip": {{ip_address}} 
+    "dst_ip": {{ip_address}} (OPTIONAL)
+```
+
+## 3.2 DASH APP DB
+
+Following diagram captures the object reference model.
+
+  ![dash-eni-vnet-obj-model](https://github.com/prsunny/DASH/blob/main/Assets/dash_object_model.png)
```

Comment:
We recommend editable .svg format, e.g. using draw.io, so that the images can be maintained down the road. Also please use a relative URL, this points to an absolute path in a personal repo dev branch. Note this image isn't in this PR yet so a merge would result in missing link.

Reply:
@prsunny Please, replace the current file images with these
![dash-hld-vnet-objects](https://user-images.githubusercontent.com/2974753/160506747-98806fba-dc5c-4c4b-9208-bfd28b2889e0.svg)
![dash-hld-eni-objects](https://user-images.githubusercontent.com/2974753/160506749-a2ad1685-7bcf-48ab-8c5e-101b4614a186.svg)
: